lib/cpp/libthrift.vcxproj

```diff
@@ -134,45 +182,53 @@
     <ConfigurationType>StaticLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
     <CharacterSet>MultiByte</CharacterSet>
+    <PlatformToolset>v142</PlatformToolset>
```

Comment:
Is `PlatformToolset` required or can Visual Studio find it automatically? Could you try to remove it (maybe with a text editor), then load the project and see if it builds correctly?

The less constraint the project is, the better, because it can remain compatible with more versions of `PlatformToolset`.

Reply:
Hi @emmenlau , I remove it and load the project but it can't builds correctly. The Visual Studio's working mechanism is: it will replace the `PlatformToolset` with your current environment's `PlatformToolset`. If your setting is not match with your current environment, it can't work.

Reply:
Dear @zeshuai007 I understand but its quite curious. The previous project did not have it, or did it? Was that an older standard?

Reply:
Dear @emmenlau , The previous project did not have it, it was generated by visual studio when I confirm to chose the `v142  PlatformToolset`. Maybe the correct way is to remove it and use what version of `PlatformToolset` is depending on the developer.

Reply:
Dear @zeshuai007 This was my suggestion to remove it an let Visual Studio automatically add it for the developer. But I understand you said this did not build correctly? Can you clarify if it can safely be removed or not?

Reply:
Dear @emmenlau , I am sorry for not clarify clearly. Visual Studio has the config menu for choose the `PlatformToolset`. Developer can config the correct `PlatformToolset` in they environment. Just like me, I config the `v142 PlatformToolset`, so it build correctly.But if i don't remove the config informoation(`<PlatformToolset>v142</PlatformToolset>`) ,it may not work for others who has the different `PlatformToolset`. So, I think remove it and config by developer again maybe a better way.Even if they occur some problem relate with this, Visual Studio also provide the output information that imply them to config it.